bundles/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/handler/TiVoHandler.java

```diff
@@ -0,0 +1,364 @@
+
+    /**
+     * Instantiates a new TiVo handler.
+     *
+     * @param thing the thing
+     */
+    public TiVoHandler(Thing thing) {
+        super(thing);
+        logger.debug("TiVoHandler '{}' - creating", getThing().getUID());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Handles the commands from the various TiVo channel objects
+        logger.debug("handleCommand '{}', parameter: {}", channelUID, command);
+
+        if (!isInitialized() || !tivoConnection.isPresent()) {
+            logger.debug("handleCommand '{}' device is not intialised yet, command '{}' will be ignored.",
+                    getThing().getUID(), channelUID + " " + command);
+            return;
+        }
+
+        TivoStatusData currentStatus = tivoConnection.get().getServiceStatus();
+        String commandKeyword = "";
+
+        String commandParameter = command.toString().toUpperCase();
+        if (command instanceof RefreshType) {
+            // Future enhancement, if we can come up with a sensible set of actions when a REFRESH is issued
+            logger.debug("TiVo '{}' skipping REFRESH command for channel: '{}'.", getThing().getUID(),
+                    channelUID.getId());
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case CHANNEL_TIVO_CHANNEL_FORCE:
+                commandKeyword = "FORCECH";
+                break;
+            case CHANNEL_TIVO_CHANNEL_SET:
+                commandKeyword = "SETCH";
+                break;
+            case CHANNEL_TIVO_TELEPORT:
+                commandKeyword = "TELEPORT";
+                break;
+            case CHANNEL_TIVO_IRCMD:
+                commandKeyword = "IRCODE";
+                break;
+            case CHANNEL_TIVO_KBDCMD:
+                commandKeyword = "KEYBOARD";
+                break;
+        }
+        sendCommand(commandKeyword, commandParameter, currentStatus);
+    }
+
+    public void setStatusOffline() {
+        this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                "Power on device or check network configuration/connection.");
+    }
+
+    private void sendCommand(String commandKeyword, String commandParameter, TivoStatusData currentStatus) {
+        if (!tivoConnection.isPresent()) {
+            return;
+        }
+
+        TivoStatusData deviceStatus = tivoConnection.get().getServiceStatus();
+        TivoStatusData commandResult = null;
+        logger.debug("handleCommand '{}' - {} found!", getThing().getUID(), commandKeyword);
+        // Re-write command keyword if we are in STANDBY, as only IRCODE TIVO will wake the unit from
+        // standby mode
+        if (deviceStatus.getConnectionStatus() == ConnectionStatus.STANDBY && commandKeyword.contentEquals("TELEPORT")
+                && commandParameter.contentEquals("TIVO")) {
+            commandKeyword = "IRCODE " + commandParameter;
```

Comment:
It's bad practice to re-assign method arguments. You could create a new local variable.

Reply:
done